third_party/move/move-model/src/model.rs

```diff
@@ -1080,32 +1080,107 @@ impl GlobalEnv {
         self.report_diag_with_filter(writer, |d| d.severity >= severity)
     }
 
+    fn cmp_diagnostic(diag1: &Diagnostic<FileId>, diag2: &Diagnostic<FileId>) -> Ordering {
```

Comment:
It would be nice if you could add a comment to explain the ordering rule a little bit in case it is not documented anywhere else.